### PR TITLE
Docs: Fixed grammar and spelling errors. Issue #20238

### PR DESCRIPTION
**Pull request by CMcCudden**
Commit Message: Docs: Fixed Grammar, spelling errors, trailing white spaces, etc.
Risk Level: Low
Docs Changes: Docs: Fixed Grammar, spelling errors, trailing white spaces, etc.
 
Signed off by: Caleb McCudden <calebmccudden@gmail.com>

### CODE REVIEW

**Comment by daixiang0** [comment]
Please fix DCO.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -200,15 +200,15 @@ Runtime guarded features may either set true (running the new code by default) i
 after a testing interval, or during the next release cycle, at the PR author's and reviewing
 maintainer's discretion. Generally all runtime guarded features will be set true when a
 release is cut. Old code paths for refactors can be cleaned up after a release and there has been
-some production run time. Old code for behavioral changes will be deprecated after six months.
+some production run-time. Old code for behavioral changes will be deprecated after six months.
```
Comment (daixiang0) [inline CONTRIBUTING.md:203]: Runtime?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -200,15 +200,15 @@ Runtime guarded features may either set true (running the new code by default) i
...
 Runtime features are set true by default by inclusion in
 [source/common/runtime/runtime_features.cc](https://github.com/envoyproxy/envoy/blob/main/source/common/runtime/runtime_features.cc)
 
 There are four suggested options for testing new runtime features:
 
 1. Create a per-test Runtime::LoaderSingleton as done in [DeprecatedFieldsTest.IndividualFieldDisallowedWithRuntimeOverride](https://github.com/envoyproxy/envoy/blob/main/test/common/protobuf/utility_test.cc)
 2. Create a [parameterized test](https://github.com/google/googletest/blob/master/docs/advanced.md#how-to-write-value-parameterized-tests)
-   where the set up of the test sets the new runtime value explicitly to
+   where the set-up of the test sets the new runtime value explicitly to
```
Comment (daixiang0) [inline CONTRIBUTING.md:211]: setup?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -238,7 +238,7 @@ temporarily reverted by setting runtime guard ``envoy.reloadable_features.schema
 * The above rules may be waived for PRs which only update docs or comments, or trivial changes to
   tests and tools (where trivial is decided by the maintainer in question).
 * In general, we should also attempt to make sure that at least one of the approvals is *from an
-  organization different from the PR author.* E.g., if Lyft authors a PR, at least one approver
+  organization different from the PR author.* E.g., if Lyft authors a PR, at least one party
```
Comment (daixiang0) [inline CONTRIBUTING.md:241]: It should be approver.

File: docs/root/configuration/listeners/network_filters/rocketmq_proxy_filter.rst [hunk 4]
```diff
@@ -63,17 +63,17 @@ following statistics:
   get_topic_route, Counter, Total getting topic route requests
   send_message_v1, Counter, Total sending message v1 requests
   send_message_v2, Counter, Total sending message v2 requests
-  pop_message, Counter, Total poping message requests
-  ack_message, Counter, Total acking message requests
+  pop_message, Counter, Total popping message requests
+  ack_message, Counter, Total packing message requests
```
Comment (zhxie) [inline docs/root/configuration/listeners/network_filters/rocketmq_proxy_filter.rst:67]: I think the original one is correct here.

**Review by daixiang0 — COMMENTED** [review]
Thanks for your contribution.

Is there a tool for it, I am happy to see it in our CI.

File: CONTRIBUTING.md (hunk 1 above)
Comment (CMcCudden) [reply CONTRIBUTING.md:203]: Most changes I made similar to this and a lot of "setup"/ "set-up"'s you'll see are adjustments for when the word is being used as a noun or as a verb. Generally, nouns will be hyphenated and verbs will be written out as two separate words. Apologies if it seems a bit nit-picky, I just figured I'd try and make the writing as consistent as possible.

**Comment by phlax** [comment]
@CMcCudden could you address above nits, and then im happy to give it a final review

**Pull request closed without merging**